src/CMSHistFunc.cc

```diff
@@ -998,7 +998,7 @@ FastTemplate CMSHistFunc::cdfMorph(unsigned idx, double par1, double par2,
               << std::endl;
 #endif
 
-  while (x >= xdisn[nx3]) {
```

Comment:
I think a better way to fix this is to restore the zero integral check on L945 in some way, and then return an empty histogram.

Reply:
Ok fine with me, I'll have a look at this. Actually, the fit does not seg fault, but still fails :) So more care is needed

Reply:
@ajgilbert I'd like to revive this pull request, since major issues encountered with the underlying morphing should be fixed now.

* In case one of yields of input templates is <= 0, an empty histogram is returned
* During testing of morphing, some boundary issues were spotted and corrected
* Debug output was improved to track down individual morphing steps and see the outcome